### PR TITLE
Attempt to set up a basic github action

### PR DESCRIPTION
**Pull request by mattmoor**
This is loosely based on the go.yaml template and this article on using GOPATH: https://laszlo.cloud/setting-gopath-in-github-actions

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.13
+      id: go
+
+    - name: Check out code onto GOPATH
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
+        path: /home/runner/work/sources-for-knative/sources-for-knative/go/src/github.com/vmware-tanzu/sources-for-knative
```
Comment (vaikas) [inline .github/workflows/go.yml:26]: I don't know enough about the syntax here but just curious why sources-for-knative is there twice?

Comment (mattmoor) [reply .github/workflows/go.yml:26]: I can try without it, but as I was beating my head against the wall there was an error message about it not being under that directory (with it there twice) 🤣

Comment (mattmoor) [reply .github/workflows/go.yml]: ok, it's running now.  If it blows up, I'll revert back to what was here.

Comment (mattmoor) [reply .github/workflows/go.yml:26]: fails without it 🤷‍♂

Comment (n3wscott) [inline .github/workflows/go.yml:26]: I bet you can change this to `$GOPATH/src/github.com/vmware-tanzu/sources-for-knative`

Comment (n3wscott) [reply .github/workflows/go.yml:26]: from:  https://github.com/mvdan/github-actions-golang, you can use

```    - name: Checkout code
      uses: actions/checkout@v2
      with:
        path: ./src/github.com/${{ github.repository }}```

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  build:
```
Comment (n3wscott) [inline .github/workflows/go.yml:11]: under build, we might want this:
```
    env:
      GOPATH: ${{ github.workspace }}
      GO111MODULE: off
```

**Comment by n3wscott** [comment]
LGTM

**Pull request merged**